// Copyright 2019 the gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package hash_name;

message HashNameRequest {
  string desired_name = 1;
  int32 ideal_hamming_distance = 2;
  int32 interesting_hamming_distance = 3;
}

message HashNameResponse {
  string secret = 1;
  string hashed_name = 2;
  int32 hamming_distance = 3;
}

service HashFinder {
  rpc Find (HashNameRequest) returns (HashNameResponse) {}
  rpc FindRange (HashNameRequest) returns (stream HashNameResponse) {}
}
